langchain/logging/base.py

```diff
@@ -0,0 +1,105 @@
+"""Base interface for logging runs."""
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Dict, List, Union
+
+from dataclasses_json import dataclass_json
+
+
+@dataclass_json
+@dataclass
+class Run:
+    id: int
+    start_time: datetime
+    end_time: datetime
+    extra: Dict[str, Any]
+    error: Dict[str, Any]
+    execution_order: int
+    serialized: Dict[str, Any]
+
+
+@dataclass_json
+@dataclass
+class LLMRun(Run):
+    prompts: Dict[str, Any]
+    response: Dict[str, Any]
+
+
+@dataclass_json
+@dataclass
+class ChainRun(Run):
+    inputs: Dict[str, Any]
+    outputs: Dict[str, Any]
+    child_runs: List[Run]
+
+
+@dataclass_json
+@dataclass
+class ToolRun(Run):
+    inputs: Dict[str, Any]
+    outputs: Dict[str, Any]
+    action: str
+    child_runs: List[Run]
+
+
+class BaseLogger(ABC):
+    """Base interface for logging runs."""
+
+    @abstractmethod
+    def log_llm_run_start(
+        self, serialized: Dict[str, Any], prompts: List[str], **extra: str
```

Comment:
is Dict[str, Any] the best form?

what goes in extra?

Reply:
Not sure what the best type is for serialized objects. I was assuming they get serialized as `json`, in which case `Dict[str, Any]` could work.

Extra is for any extra fields that you might want to store. Left it in there for added flexibility, but it's not used